### PR TITLE
test: add edge case tests for number comparison in compare utility

### PR DESCRIPTION
**Pull request by virajsabhaya23**
missing edge cases for comparison between diff sets of number

### CODE REVIEW

File: packages/@ember/utils/tests/compare_test.js [hunk 1]
```diff
@@ -84,5 +84,15 @@ moduleFor(
       assert.equal(compare('b', zero), 0, 'Second item comparable - returns  0 (negated)');
       assert.equal(compare('c', one), -1, 'Second item comparable - returns  1 (negated)');
     }
+    
+    ['@test comparing numbers with edge cases'](assert) {
```
Comment (NullVoxPopuli) [inline packages/@ember/utils/tests/compare_test.js:88]: I think the above tests cover these 🤔

Comment (kategengler) [reply packages/@ember/utils/tests/compare_test.js:88]: Yes, they are all covered above.
